models/models.js

```diff
@@ -64,10 +64,40 @@ function checkArticleIdExists(articleId) {
     });
 }
 
+function insertCommentByArticleId(articleId, newComment) {
+  return checkIfUserExists(newComment.username)
+    .then(() => {
+      return db
+        .query(
+          ` INSERT INTO comments (body, author, article_id)
+            VALUES ($1, $2, $3)
+            RETURNING * ;`,
+          [newComment.body, newComment.username, articleId]
+        )
+        .then(({ rows }) => {
+          return rows[0];
+        });
+    })
+    .catch((err) => {
```

Comment:
This catch isn't needed as the checkIfUserExists already has a promise reject. As you're doing two promises and want them both to reject if checkIfUserExists rejects, I'd recommend a refactor to a promise.all.